packages/nitro-protocol/docs/app-devs/quick-start-dapp.md

```diff
@@ -41,3 +41,21 @@ window.channelProvider.enable();
 If everything is setup correctly, you should see the statechannels wallet UI:
 
 ![](assets/wallet-ui.png)
+
+## App <-> Wallet security
+
+Having the wallet served in an iFrame from a different domain to the app is an important component from a security standpoint. It means app <-> wallet communication is cross-origin and therefore disabled by default, but capable of being selectively enabled on a per-origin (per-app) basis. Even after being enabled, the app can only send and receive messages to the wallet, and cannot otherwise control it.
```

Comment:
"is cross-origin and therefore disabled by default" - is this true? Doesn't `postMessage` work by default?

Reply:
I think the importance of the `iFrame` is that the app doesn't have access to the wallet's code or storage, due to the way that the browser sandboxes each domain.

Reply:
I think you're right about the main importance of the iFrame, and I think whether _communication_ from app to wallet is allowed is secondary to it. 

> "is cross-origin and therefore disabled by default" - is this true? Doesn't `postMessage` work by default?

It doesn't work by default https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage. In our current implementation we tag incoming messages by `origin`, so that we can e.g. associate channels and budgets with that domain:

https://github.com/statechannels/statechannels/blob/ddee45789b0b0df95c180b997c165cce0c8559a6/packages/xstate-wallet/src/index.ts#L42-L55

Outgoing messages have `targetOrigin: '*'` currently, so I think there's a chance they could be sniffed. 

I'll try and describe this a little more accurately.

Reply:
https://github.com/statechannels/statechannels/pull/2356/commits/f893df1fec98f0adff01e44f81be14b6657fb6de